### PR TITLE
Added a Jenkinsfile

### PR DESCRIPTION
**Pull request by AdamSaleh**
## Motivation

We want to have PR builds on Jenkins
JIRA: https://issues.jboss.org/browse/AGDROID-704

## Description

Setup slaves and configure Jenkins to run on PR. Make a basic sanity build.

## Progress

- [x] Added sdk volume to openshift project
- [x] Setup organization to check PRs of aerogear-android-sdk
- [x] Make the PR actually check

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    cloud: "openshift", 
+    containers: [androidSdkSlave],
+    volumes: [persistentVolumeClaim(mountPath: '/opt/android-sdk-linux', claimName: 'android-sdk', readOnly: true)]) {
+  node ("android-sdk") {
+    stage("Checkout") {
+      checkout scm
+    }
+    
+    stage ("Build") {
+      sh " ./gradlew :core:assembleDebug"
```
Comment (danielpassos) [inline Jenkinsfile]: `assembleDebug` don't run any test. You need to use `testDebug`

Comment (danielpassos) [reply Jenkinsfile]: You also should not do it only on _core_

**Comment by danielpassos** [comment]
@AdamSaleh I have no problem to add Jenkins build here, but I don't wanna it block any PR. What is the point to add Jenkins build instead of use a community (travis-ci or circle-ci)?

**Comment by AdamSaleh** [comment]
@danielpassos there are two points
1. dogfooding, we want our customers to use aerogear-digger, so we should use the aerogear-digger
2. greater control, i.e. it should be easier to add e2e integration tests for our sdks when running on openshift already, it should be easier to add osX builders/runners (I think travis might support those?), it would be easier to do more parallel tests (free resources on travis/circleci seem quite limited)

But if we come to a place, where maintainance burden is bigger than advantages of rolling our own stuff, by all means, we should use something else. Preferably with a budget ;-)

**Comment by secondsun** [comment]
@AdamSaleh How do I test this?

**Comment by danielpassos** [comment]
@secondsun I think it is self-tested by Jenkinks. You can see the build on the PR https://jenkins-wendy.ci.feedhenry.org/job/aerogear/job/aerogear-android-sdk/job/PR-20/8/display/redirect

**Pull request merged**